power_grid_model_c/power_grid_model/include/power_grid_model/prepare_calculate.hpp

```diff
@@ -0,0 +1,120 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+
+#include "topology.hpp"
+
+#include "common/common.hpp"
+
+#include "math_solver/math_solver_dispatch.hpp"
+
+#include "main_core/main_model_type.hpp"
+#include "main_core/math_state.hpp"
+#include "main_core/topology.hpp"
+#include "main_core/y_bus.hpp"
+
+namespace power_grid_model {
+struct SolverPreparationContext {
+    main_core::MathState math_state;
+    Idx n_math_solvers{0};
```

Comment:
Is it required to keep n_math_solvers?
Is it ever not equivalent to `math_state.topology.size()`?